src/corell/src/lib.rs

```diff
@@ -123,6 +135,8 @@ pub struct Device<R: Resources, F: Factory<R>, Q: CommandQueue> {
     pub transfer_queues: Vec<TransferQueue<Q>>,
     pub heap_types: Vec<HeapType>,
     pub memory_heaps: Vec<u64>,
+    /// Device capabilities
+    pub caps: Capabilities,
```

Comment:
Could we already have access to these capabilities from `Adapter`?
So users could have more control about which device to open beside the current `AdapterInfo`.